c7n/resources/awslambda.py

```diff
@@ -255,6 +255,45 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'KMSKeyArn'
 
 
+@AWSLambda.action_registry.register('enable-xray-tracing')
+class LambdaEnableXrayTracing(Action):
+    """
+        This action allows for enable Xray tracing to Active
+       :example:
+       .. code-block:: yaml
+           actions:
+             - type: enable-xray-tracing
+    """
+
+    schema = type_schema('enable-xray-tracing')
+    permissions = ("lambda:UpdateFunctionConfiguration",)
+
+    def process(self, resources):
+        """
+            Enables the Xray Tracing for the function.
+
+            Args:
+                resources: AWS lamdba resources
+            Returns:
+                None
+        """
+        client = local_session(self.manager.session_factory).client('lambda')
+
+        for resource in resources:
+            if resource["TracingConfig"]["Mode"] != "Active":
+                function_name = resource["FunctionName"]
+                self.log.info(f"Enabling Xray tracing for lambda {function_name}")
+                try:
+                    client.update_function_configuration(
+                        FunctionName=function_name,
+                        TracingConfig={
+                            'Mode': 'Active'
+                        }
+                    )
+                except Exception as ex:
```

Comment:
we should catch `ResourceNotFoundException` and continue, and retry on `TooManyRequestsException` and `ResourceConflictException` otherwise we should raise

Reply:
Your right! updated code to handle above exceptions.